actor/v2action/service_instance_summary.go

```diff
@@ -67,31 +67,39 @@ func (actor Actor) GetServiceInstanceSummaryByNameAndSpace(name string, spaceGUI
 }
 
 func (actor Actor) GetServiceInstancesSummaryBySpace(spaceGUID string) ([]ServiceInstanceSummary, Warnings, error) {
-	serviceInstances, warnings, err := actor.CloudControllerClient.GetSpaceServiceInstances(
-		spaceGUID,
-		true)
+	spaceSummary, warnings, err := actor.CloudControllerClient.GetSpaceSummary(spaceGUID)
 	allWarnings := Warnings(warnings)
+	if err != nil {
+		return []ServiceInstanceSummary{}, allWarnings, err
+	}
 
-	log.WithField("number_of_service_instances", len(serviceInstances)).Info("listing number of service instances")
-
-	var summaryInstances []ServiceInstanceSummary
-	for _, instance := range serviceInstances {
-		serviceInstanceSummary, summaryInfoWarnings, summaryInfoErr := actor.getSummaryInfoCompositeForInstance(
-			spaceGUID,
-			ServiceInstance(instance),
-			false)
-		allWarnings = append(allWarnings, summaryInfoWarnings...)
-		if summaryInfoErr != nil {
-			return nil, allWarnings, summaryInfoErr
-		}
-		summaryInstances = append(summaryInstances, serviceInstanceSummary)
+	services, warnings, err := actor.CloudControllerClient.GetServices()
```

Comment:
Done

Reply:
Yes, there is a reason for using `/v2/services` instead of `/v2/spaces/:guid/services`.

`/v2/spaces/:guid/services` returns only the services which are currently visible to your space. In the case where someone has disabled the service access to that service offering, but you still have a service instance in your space `/v2/spaces/:guid/services` won't return that service while `/v2/services` will. This results in missing broker name in case the service offering is disabled for your org. Hope the following snippet makes it more clear.

```
→ cf enable-service-access overview-service
Enabling access to all plans of service overview-service for all orgs as admin...
OK

→ cf curl /v2/services
{
   "total_results": 1,
   "total_pages": 1,
   "prev_url": null,
   "next_url": null,
   "resources": [
    ...
   ]
}

→ cf curl /v2/spaces/1aff19b7-8185-49ab-a4bd-e79dc50f943f/services
{
   "total_results": 1,
   "total_pages": 1,
   "prev_url": null,
   "next_url": null,
   "resources": [
   ...
   ]
}

→ cf disable-service-access overview-service
Disabling access to all plans of service overview-service for all orgs as admin...
OK

→ cf curl /v2/spaces/1aff19b7-8185-49ab-a4bd-e79dc50f943f/services
{
   "total_results": 0,
   "total_pages": 1,
   "prev_url": null,
   "next_url": null,
   "resources": [] // NOTE the empty resources
}

→ cf curl /v2/services
{
   "total_results": 1,
   "total_pages": 1,
   "prev_url": null,
   "next_url": null,
   "resources": [
    ...
   ]
}
```

One option for this workflow to be simplified further and the call to `/v2/services` omitted is if we have the broker name as part of the summary response.
https://www.pivotaltracker.com/story/show/163455079
Another is to change the `/v2/space/:guid/services` behaviour, but I'm not sure if it won't affect something else.

I think for now we should stick with the call to `/v2/services`, since it won't require CC changes and improve it in future.